cachi2/core/package_managers/gomod.py

```diff
@@ -582,7 +582,37 @@ def _resolve_gomod(
         # Make Go ignore the vendor dir even if there is one
         go_list.extend(["-mod", "readonly"])
 
-    main_module_name = _run_gomod_cmd([*go_list, "-m"], run_params).rstrip()
+    go_list_modules = _run_gomod_cmd([*go_list, "-m", "-json"], run_params).rstrip()
+    module_list = list(load_json_stream(go_list_modules))
+    raw_main_module = module_list.pop(0)
+    main_module_name = raw_main_module["Path"]
+
+    def get_relative_path_workspace_to_main_module(path1: Path, path2: Path) -> str:
+        """
+        Return relative path from workspace module to main module.
```

Comment:
Are relative paths fine wrt/ SBOMs? Don't we need to canonicalize the resulting path?

Reply:
Proper path is determined [here](https://github.com/containerbuildsystem/cachi2/blob/main/cachi2/core/package_managers/gomod.py#L468), that's why we create workspace module as `ParsedModule` with `replace`.